iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -190,6 +180,78 @@ public void commitDropTable(org.apache.hadoop.hive.metastore.api.Table hmsTable,
     }
   }
 
+  @Override
+  public void preAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, EnvironmentContext context)
+      throws MetaException {
+    HiveMetaHook.super.preAlterTable(hmsTable, context);
+    catalogProperties = getCatalogProperties(hmsTable);
+    try {
+      icebergTable = Catalogs.loadTable(conf, catalogProperties);
+    } catch (NoSuchTableException nte) {
+      // If the iceberg table does not exist, and the hms table is external and not temporary, not acid,
+      // and not bucketed, we will create it in commitAlterTable
+      StorageDescriptor sd = hmsTable.getSd();
+      generateMetadata = MetaStoreUtils.isExternalTable(hmsTable) && !hmsTable.isTemporary() &&
+          !(sd.getNumBuckets() > 0) && !AcidUtils.isTransactionalTable(hmsTable);
```

Comment:
Can we not migrate a bucketed table to Iceberg? Would the table just lose its bucketed property but otherwise the migration would succeed, or do we run into some errors because of the way the execution plan is put together differently for bucketed tables?

Reply:
It turns out we can :). Removed this condition.